api/v1alpha1/nicdevice_types.go

```diff
@@ -20,22 +20,37 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+type NicDeviceConfigurationSpec struct {
+	ResetToDefault bool `json:"resetToDefault,omitempty"`
+	// TODO add NicConfigurationTemplateSpec once #2 is merged
+}
 
 // NicDeviceSpec defines the desired state of NicDevice
 type NicDeviceSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	Configuration *NicDeviceConfigurationSpec `json:"configuration,omitempty"`
+}
+
+type NicDevicePortSpec struct {
+	Pci              string `json:"pci"`
+	NetworkInterface string `json:"networkInterface,omitempty"`
+	RdmaInterface    string `json:"rdmaInterface,omitempty"`
+}
 
-	// Foo is an example field of NicDevice. Edit nicdevice_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+type NicDeviceConditionSpec struct {
+	Type   string `json:"type"`
+	Status string `json:"status"`
 }
 
 // NicDeviceStatus defines the observed state of NicDevice
 type NicDeviceStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	Node            string                   `json:"node"`
+	Type            string                   `json:"type"`
+	SerialNumber    string                   `json:"serialNumber"`
+	PartNumber      string                   `json:"partNumber"`
+	Psid            string                   `json:"psid"`
```

Comment:
These are initials of Product Serial Identification. do we want to name it PSID ? or leave as is ??

Reply:
I don't have a preference, let's capitalize it